submissions/AnnaStoyano/Friends-app/index.js

```diff
@@ -0,0 +1,194 @@
+    }
+
+    filterByName() {
+        this.filterByNameAlphabetically();
+        this.filterByNameNonAlphabetically();
+    }
+
+    filterByNameAlphabetically() {
+        const app = this;
+        const nameAlphabeticallyBtn = document.querySelector('.name-alphabetically');
+        nameAlphabeticallyBtn.addEventListener('click', function () {
+            app.dipslayCards = app.dipslayCards.sort((a, b) => a.name > b.name ? -1 : 1);
+            app.render();
+        });
+    }
+
+    filterByNameNonAlphabetically() {
+        const app = this;
+        const nameNonAlphabeticallyBtn = document.querySelector('.name-nonalphabetically');
+        nameNonAlphabeticallyBtn.addEventListener('click', function () {
+            app.dipslayCards = app.dipslayCards.sort((a, b) => a.name > b.name ? 1 : -1);
+            app.render();
+        });
+    }
+
+    filterByAge() {
+        const app = this;
+        const ageBtn = document.querySelector('.age_btn');
+        ageBtn.addEventListener('click', function () {
+            if (!ageBtn.classList.contains('LowToHigh')) {
+                app.dipslayCards = app.dipslayCards.sort((a, b) => a.age > b.age ? -1 : 1);
+                ageBtn.classList.add('LowToHigh');
+            } else {
+                app.dipslayCards = app.dipslayCards.sort((a, b) => a.age > b.age ? 1 : -1);
+                ageBtn.classList.remove('LowToHigh');
+            }
+            app.render();
+        });
+    }
+
+    filterByGender() {
+        const app = this;
+        const manCheckbox = document.querySelector('.checkbox-input.man');
+        const womanCheckbox = document.querySelector('.checkbox-input.woman');
+        const genderWrapper = document.querySelector('.menu_item-gender');
+        genderWrapper.addEventListener('click', function () {
+            if (womanCheckbox.checked && manCheckbox.checked) {
+                app.dipslayCards = app.cards;
+            }
+            if (manCheckbox.checked && !womanCheckbox.checked) {
+                app.dipslayCards = app.dipslayCards.filter(card => card.user.gender == 'male');
+            }
+            if (womanCheckbox.checked && !manCheckbox.checked) {
+                app.dipslayCards = app.dipslayCards.filter(card => card.user.gender == 'female');
+            }
+            if (!womanCheckbox.checked && !manCheckbox.checked) {
+                app.dipslayCards = app.cards;
+            }
+            app.render();
+        });
+    }
+
+    filterByCity() {
+        const app = this;
+        const cityInput = document.querySelector('.text-input--city');
+        const cityWrapper = document.querySelector('.menu_item-city');
+        cityWrapper.addEventListener('click', function () {
+            app.prevDisplay = app.dipslayCards;
+        });
+        cityInput.addEventListener('input', function ({target}) {
+            app.dipslayCards = app.dipslayCards.filter(card => card.user['location']['city'].includes(target.value));
```

Comment:
No need for array notation, use dots instead.

Also, I think `String#includes` or `String#startsWith` will fit better